### PR TITLE
[MRG+1] DOC Updated plot_date to NumPy/SciPy style

### PR DESCRIPTION
**Pull request by klaragerlei**
Hello,

We updated the documentation of plot_date to NumPy/SciPy style.

Thanks,
Michael, Klara

@NelleV

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -1408,59 +1408,66 @@ def plot(self, *args, **kwargs):
     def plot_date(self, x, y, fmt='o', tz=None, xdate=True, ydate=False,
                   **kwargs):
         """
-        Plot with data with dates.
+	A plot with data that contains dates.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: These tabs need to be replaced with the correct number of spaces.

**Comment by QuLogic** [comment]
It appears you have not [set the config for your git client](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity). All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1408,59 +1408,66 @@ def plot(self, *args, **kwargs):
...
+    	-------
+    	lines
 
-        %(Line2D)s
 
-        .. seealso::
+	See Also
+    	--------
+    	mod : matplotlib.dates for helper functions
+        func : matplotlib.dates.date2num
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There is a strange indentation here.
In addition, we need to revert back to the old syntax: :func:`~matplotlib.dates.date2num` for proper formatting.

**Comment by NelleV** [comment]
Hi Michael & Klara,

The code currently has a mixture of tabs and space, due to gedit not being configured correctly. I can drop by at some point today to help you fix that.
line 1459 also appears to be too long.
Once this is fixed, I think looks great!

Thanks,
Nelle

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
...
-
-        Call signature::
-
-           plot_date(x, y, fmt='bo', tz=None, xdate=True,
-                     ydate=False, **kwargs)
-
-        Similar to the :func:`~matplotlib.pyplot.plot` command, except
+        A plot with data that contains dates.
+        
+        Similar to the :func:~matplotlib.pyplot.plot command, except
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: The backticks should not have been removed here.

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
...
         the *x* or *y* (or both) data is considered to be dates, and the
         axis is labeled accordingly.
 
         *x* and/or *y* can be a sequence of dates represented as float
         days since 0001-01-01 UTC.
 
-        Keyword arguments:
+        Note if you are using custom date tickers and formatters, it
+        may be necessary to set the formatters/locators after the call
+        to meth : plot_date since meth : plot_date will set the
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: The backticks should not have been removed here and the spaces around the colons are not necessary.

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
...
+        mod : matplotlib.dates for helper functions
+        :func:~matplotlib.dates.date2num
+        :func:~matplotlib.dates.num2date
+        :func:~matplotlib.dates.drange for help on creating the required
+        floating point dates
+
+
+        Other Parameters
+        ----------------
+        kwargs : class : matplotlib.lines.Line2D properties : %(Line2D)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Line2D` gets expanded to several other parameters; it should be on a line of its own.

**Comment by QuLogic** [comment]
Please remember to correct the name and committer on the first commit.

**Comment by NelleV** [comment]
@QuLogic FYI, we did it, but it requires rebasing and some quite complex operations. I don't think that should be a requirement for contributing to matplotlib (in addition, some people may want to stay anonymous).

**Comment by NelleV** [comment]
@klaragerlei There is still a couple of problems: I can help you with those during a break or the tutors consulting hours tonight if you'd like.

**Comment by klaragerlei** [comment]
That would be great @Nelle, thank you!

File: lib/matplotlib/axes/_axes.py [hunk 6]
```diff
@@ -1408,59 +1408,68 @@ def plot(self, *args, **kwargs):
...
         """
-        Plot with data with dates.
-
-        Call signature::
-
-           plot_date(x, y, fmt='bo', tz=None, xdate=True,
-                     ydate=False, **kwargs)
-
+        A plot with data that contains dates.
+
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: There is still some whitespace on this line.

**Comment by QuLogic** [comment]
It appears that you were able to take care of the rebasing? Just a small note that your name appears differently in some of the commits.

CI isn't happy right now, but it is raising a cryptic error. We will have to try a local build to figure out what is wrong.

File: lib/matplotlib/axes/_axes.py [hunk 7]
```diff
@@ -1408,59 +1408,68 @@ def plot(self, *args, **kwargs):
...
+        :func:`~matplotlib.dates.date2num`
+        :func:`~matplotlib.dates.num2date`
+        :func:`~matplotlib.dates.drange` for help on creating the required
+        floating point dates
+
+
+        Other Parameters
+        ----------------
+        kwargs : class : matplotlib.lines.Line2D
+        properties : %(Line2D)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Line2D` gets expanded into several parameter listings; it should be on its own line. (And likely needs to be updated to match this formatting.)

**Comment by NelleV** [comment]
Thanks a lot! This is looks great:

plot_date

**Comment by QuLogic** [comment]
Are you okay with the some commits using your GitHub username and not your real name like the rest of them? Also, all but the second-last commit use the email of @ClaireSophie but the name is @klaragerlei. Please double-check that you will be attributed correctly.

**Comment by NelleV** [comment]
This PR has been done as pair coding, so it is going to be hard to attribute it properly.

**Comment by klaragerlei** [comment]
@QuLogic I think I managed to fix the double account problem, and it's all correct now, thank you for noticing.

**Comment by NelleV** [comment]
@klaragerlei Thanks!

**Comment by tacaswell** [comment]
backported to v2.x as 2ddc50e

**Comment by QuLogic** [comment]
There were still two emails with the same name used for different commits; are they the same person?

**Comment by klaragerlei** [comment]
Yes, both of them are mine, sorry, one of them is a university email
address, and the other one is personal. I realized that the university
email address might not be permanent enough, so I switched to using the
other one, but I used that email address ages ago on github for
contributing to a funny robot game, but I completely forgot (this is how
the other account appeared in my commit when I used the personal email in
the configurations), but now I deleted the old account and added both
emails to the same account, so I hope it's all fine now.

**Pull request merged**